catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -13,261 +13,174 @@
+import com.google.common.io.FileBackedOutputStream;
 import ddf.catalog.data.BinaryContent;
 import ddf.catalog.data.Metacard;
 import ddf.catalog.data.Result;
 import ddf.catalog.data.impl.BinaryContentImpl;
-import ddf.catalog.data.impl.MetacardImpl;
-import ddf.catalog.operation.ResourceRequest;
-import ddf.catalog.operation.ResourceResponse;
 import ddf.catalog.operation.SourceResponse;
-import ddf.catalog.operation.impl.ResourceRequestById;
-import ddf.catalog.resource.Resource;
-import ddf.catalog.resource.ResourceNotFoundException;
-import ddf.catalog.resource.ResourceNotSupportedException;
 import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.MetacardTransformer;
 import ddf.catalog.transform.QueryResponseTransformer;
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
 import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
-import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.ObjectOutputStream;
 import java.io.Serializable;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
-import java.util.HashMap;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.zip.ZipEntry;
-import java.util.zip.ZipInputStream;
 import java.util.zip.ZipOutputStream;
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
 import org.apache.commons.collections.CollectionUtils;
-import org.apache.commons.collections.MapUtils;
-import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang.StringUtils;
-import org.codice.ddf.configuration.SystemBaseUrl;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class ZipCompression implements QueryResponseTransformer {
 
-  public static final String METACARD_PATH = "metacards" + File.separator;
-
   private static final Logger LOGGER = LoggerFactory.getLogger(ZipCompression.class);
 
-  private CatalogFramework catalogFramework;
+  private static final String METACARD_PATH = "metacards" + File.separator;
+
+  private static final String TRANSFORMER_ID = "transformerId";
+
+  private static final int BUFFER_SIZE = 4096;
 
-  private JarSigner jarSigner;
+  private List<ServiceReference> metacardTransformers;
 
-  public ZipCompression(JarSigner jarSigner) {
-    this.jarSigner = jarSigner;
+  private BundleContext bundleContext;
+
+  private static MimeType mimeType;
+
+  static {
+    try {
+      mimeType = new MimeType("application/zip");
+    } catch (MimeTypeParseException e) {
+      LOGGER.warn("Failed to apply mimetype application/zip");
+    }
+  }
+
+  public ZipCompression(BundleContext bundleContext, List<ServiceReference> metacardTransformers) {
+    this.bundleContext = bundleContext;
```

Comment:
I talked with @tbatie about this. We want to avoid directly using the bundle context like this as much as possible.

There was a CDM PR that went into DDF recently that demonstrates how to get a service reference list (https://github.com/codice/ddf/blob/2.13.x/catalog/transformer/catalog-transformer-bootflag/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L21) and use it (https://github.com/codice/ddf/pull/4086/files#diff-811b42b33f2680e4ed9b8ef0711f27ddR68) instead of having to access the bundle context directly.

Reply:
So the issue here is that I need to be able to lookup the ID of the transformer. If I use a `SortedServiceList` then I would be given a `List<MetacardTransformers>` and there would be no way to lookup which metacard transformer to use. Hence why I have to go from 

`List<ServiceReference>` -> `MetacardTransformer` using `BundleContext`